docs/upgrade-to-v6.md

```diff
@@ -0,0 +1,507 @@
+
+Examples of using alternative paths are as shown below.
+
+1. List children from a user's drive.
+```java
+// Get the user's driveId 
+Drive driveItem = graphClient.me().drive().get();
+String driveId = driveItem.getId();
+// List children of a DriveItem in the Drive
+DriveItemCollectionResponse response = graphClient.drives().byDriveId(driveId).items().byDriveItemId("Item Id").children().get();
+List<DriveItem> children = response.getValue();
+```
+>NOTE: /drives/{drive-id}/root is a shorthand for /drives/{drive-id}/items/root, so 'Item Id' can be replaced with 'root' to make a call to get the root folder and subsequently its children as shown below.
+```java
+// List children in the drive by replacing 'Item Id' with 'root'
+DriveItemCollectionResponse response = graphClient.drives().byDriveId(driveId).items().byDriveItemId("root").children().get();
+List<DriveItem> children = response.getValue();
+// Or more simply
+DriveItem root = graphClient.drives().byDriveId(driveId).root().get();
+List<DriveItem> children = root.getChildren();
+```
+>NOTE: 'Item Id' can also be replaced with 'root:/{file-path}' to make a call to get a specific file or folder and subsequently its children as shown below. For more information on alternative paths see documentation linked above.
+
+2. List children from a site's drive.
+```java 
+// Get the site's driveId
+Drive siteDrive = graphClient.sites().bySiteId("Site Id").drive().get();
+String siteDriveId = siteDrive.getId();
+// List children of a DriveItem in the site's drive
+DriveItemCollectionResponse response = graphClient.drives().byDriveId(siteDriveId).items().byDriveItemId("Item Id").children().get();
+// See note above for using 'root' as the 'Item Id'
+```
+
+3. List children from a group's drive.
+```java 
+// Get the group's drive
+Drive groupDrive = graphClient.groups().byGroupId("Group Id").drive().get();
+String groupDriveId = groupDrive.getId();
+// List children of a DriveItem in the group's drive
+DriveItemCollectionResponse response = graphClient.drives().byDriveId(groupDriveId).items().byDriveItemId("Item Id").children().get();
+// See note above for using 'root' as the 'Item Id'
+```
+
+#### Upload a Small File with conflictBehavior Set
+
+To upload a small file (size should not exceed 4mb according to the [docs](https://learn.microsoft.com/en-us/graph/api/driveitem-put-content?view=graph-rest-1.0&tabs=http)) and set the `conflictBehavior` [instance attribute](https://learn.microsoft.com/en-us/graph/api/resources/driveitem?view=graph-rest-1.0#instance-attributes) you'll need to do it this way:
+
+```java
+RequestInformation requestInformation = graphClient.drives().byDriveId("Drive-Id").items().byDriveItemId("root:/MediaMeta.xml").content().toPutRequestInformation(file);
+// 'file' should be an inputStream
+URI uriIncludesConflictBehavior = new URI(requestInformation.getUri().toString()+"?@microsoft.graph.conflictBehavior=rename");
```

Comment:
can you create an issue in the metadata repo if one doesn't exist already so we add an additional annotation where this is supported and this query parameter is emitted please?

Reply:
The conflictBehavior option is referred to as an 'InstanceAttribute' and not an QueryParameter in the documentation, this could be why it is not emitted as a query parameter. Would this set the precedent that these instance attributes should be treated as QueryParams and thus belong in the query parameter options list? Would this be across the languages? And would this be more of a decision for Kiota or only for our graph repo?

Reply:
Kiota doesn't know anything about OData conventions.
This would be for openapi.net.odata and for the metadata repos (I suggest we start in the metadata repo so we can identify the right annotation).
I'm not aware of a way to document instance annotations in CSDL today but it might be worth exploring.
As for the query parameter itself, we already have a way to document those through annotations.
Start a discussion there, we'll eventually figure it out

Reply:
Ramses has created the issue at https://github.com/microsoftgraph/msgraph-metadata/issues/560